jerry-core/ecma/base/ecma-helpers-conversion.c

```diff
@@ -1011,18 +1010,23 @@ ecma_double_to_binary_floating_point (double val, /**< ecma number */
   {
     while (integer_part > 0.0)
     {
-      integer_part_buffer[char_cnt++] = (lit_utf8_byte_t) ((int) fmod (integer_part, 10.0) + '0');
+      buffer_p[num_of_digits - 1 - char_cnt++] = (lit_utf8_byte_t) ((int) fmod (integer_part, 10.0) + '0');
       integer_part = floor (integer_part / 10.0);
     }
-
-    for (i = 0; i < char_cnt; i++)
-    {
-      buffer_p[i] = integer_part_buffer[char_cnt - i - 1];
-    }
   }
   else
   {
-    ecma_double_to_ascii (val, buffer_p, &char_cnt);
+    if (num_of_digits <= 21)
```

Comment:
I think we can use the `else if` chain here, too. I.e.,
```c
if (fabs (integer_part) < EPSILON)
else if (integer_part < 10e16)
else if (num_of_digits <= 21)
else
```
(instead of nesting an `if/else` in the final `else`).

BTW, now that we have `num_of_digits` in this function, do we need `integer_part < 10e16` in the second `if` or can we have `num_of_digits <= 16`? Integer comparisons should be faster than FP, in general.

Reply:
That's a good idea.